### DISCUSSION TITLE
Undecorated main window

### DISCUSSION THREAD

**Discussion by WerWolv** [ocornut/imgui#3822] [Old/Archived discussions]
Hi, Is it possible to remove the default window decoration (titlebar, etc) of the main window somehow and let the moving and resizing be handled by ImGui itself as it does with any subwindow with the multi-viewport feature enabled?

My usecase would be to make my own title bar but I'd like to avoid having to manually implement dragging, resizing and everything when ImGui already does it.

**Comment by ocornut** [comment]
Hello,

It's been a commonly requested thing to support a "no main viewport" mode in multi-viewports. Curiously in the past 3-6 months everyone started requesting that (but we discussed it earlier) so I guess it'll happen eventually. Some of the recent changes in 1.81 were designed as a small step toward it.

It's not technically supported yet, but it seems like a common hack/workaround has been to create a 1x1 window with no decoration and minimize/hide it. I haven't tried that tho.

Ref e.g.
https://github.com/ocornut/imgui/issues/3350
https://github.com/ocornut/imgui/issues/3680

**Reply by WerWolv** [reply]
Nice that worked. Thanks a lot!

**Reply by WerWolv** [reply]
![image](https://user-images.githubusercontent.com/10835354/108251039-41810500-7157-11eb-95e3-253a56400d70.png)

It looks pretty nice in my opinion. However it turned out pretty impractical since turning off window decorations also disables all window snapping and everything provided by Windows

**Comment by ocornut** [comment]
Its a known and fairly annoying issue in winapi. There are various repository with workaround eg
https://github.com/melak47/BorderlessWindow
Would need to investigate them thoroughly see what works and what can be included as a feature of the win32 backend
